Docs/Specifics/Vision.md

```diff
@@ -2,64 +2,113 @@
+
+### Object Detection
+
+Sometimes we want to be able to detect other things on the field that aren't AprilTags.
+This could include game pieces, other robots, and more.
+One way of accomplishing this is through tuning HSV thresholds to only pick up on a certain color, such as the orange of a Crescendo note.
+Another way is contour filtering, which looks for the outlines of a shape like a circle or a triangle, such as a cone from Charged Up.
+
+<img src="../../Assets/ShapeDetection.webp" alt="Example of detecting a triangular object" width="600"/>
+
+A more complicated system is through machine learning models that have been specially trained on the objects we want to detect, such as game pieces or other robots.
+Due to the significant processing power that this requires, in FRC this is typically done on the Rockchip Neural Processing Unit (NPU), which is on the Orange Pi 5 and similar coprocessors.
+While we haven't implemented this, the greater FRC community has developed several models and other resources.
+In 2024, Photonvision also ships with a pretrained model for notes.
 
 ### PhotonVision
 
-Photonvision is an app and library that gets vision data from the robot for a variety of applications.
-It makes it a lot easier to interface with the camera and also has plenty of utility methods to get information like distance and position, as well as having some built in pose estimator features and more.
+Photonvision is a community, open source library that gets vision data from the robot for a variety of applications.
+It simplifies interfacing with the camera and also has plenty of utility methods to get information like distance and position, as well as having some built in pose estimator features and more.
 
 ### Limelight
 
 Limelight (LL) is a camera and processor system for FRC designed to make vision easy and simple.
-We used a Limelight 2+ with Limelight's software in 2022 for RR tape detection, and a LL2+ with Photonvision software in 2023 for Apriltag detection.
-The LL2+ was not satisfactory for apriltag detection for us, and we will likely move on to different hardware in the near future.
+We used a Limelight 2+ with Limelight's software in 2022 for retroreflective tape detection, and a LL2+ with Photonvision software in 2023 for Apriltag detection.
+The LL2+ was not satisfactory for Apriltag detection for us, so we switched to the...
+
+### Orange Pi + Arducam
+
+While we have the roboRIO to run much of our code on the robot, it is not capable of handling vision processing at a usable speed.
+Instead, we run this on a coprocessor, which is a second, faster computer also onboard the robot.
+For vision, we've been using the Orange Pi 5, a single board computer which is quite similar to Raspberry Pis you may have seen elsewhere.
+This is what Photonvision runs on.
+
+![Orange Pi 5](/Assets/OrangePi.jfif)
+
+Of course, we also need a camera.
+We've been using Arducam USB cameras, though other USB cameras also work.
```

Comment:
mention a few models we've used (and results w those) and any other common ones that we havent